homeassistant/config_entries.py

```diff
@@ -266,13 +268,14 @@ class ConfigFlowResult(FlowResult, total=False):
     version: int
 
 
-class ConfigEntry:
+class ConfigEntry(Generic[_DataT]):
     """Hold a configuration entry."""
 
     entry_id: str
     domain: str
     title: str
     data: MappingProxyType[str, Any]
```

Comment:
Is the goal for the new shared_data to cover the existing usage of the `data` field?
Is there a transition plan?
Has an alternative been considered to transition this field from being a mapping type to be any data holding type? Having 2 fields in parallel can add confusion to users (there is no documentation on which one should be used) and might be better to transition the current one rather than adding a new field and having 2 ways of doing the same thing.

Reply:
> Is the goal for the new shared_data to cover the existing usage of the data field?

No. They are not the same.

> Is there a transition plan?

There is nothing to transition.